igc_lib.py

```diff
@@ -71,7 +73,162 @@ def _rawtime_float_to_hms(timef):
+               
+        start_hours, start_minutes = start_time.split(':')
+        start_time = int(start_hours) * 3600 + int(start_minutes) * 60
+        end_time = 23*3600 + 59*60 + 59  #default end_time of 23:59
+        
+        #create a dictionary of names and a list of longitudes and latitudes
+        #as the waypoints co-ordinates are stored separate to turnpoint details
+        coords = defaultdict(list)
+        
+        for point in wpoints:    
+            name = point.getAttribute("name")
+            longitude = float(point.getAttribute("longitude"))
+            latitude = float(point.getAttribute("latitude"))
+            coords[name].append(longitude)
+            coords[name].append(latitude)
+
+            
+        # create list of turnpoints    
+        turnpoints = []
+        for point in tpoints:
+            lat = coords[point.getAttribute("name")][1]
+            lon = coords[point.getAttribute("name")][0]
+            radius = float(point.getAttribute("radius"))/1000
+            
+            if point == tpoints[0]: #if it is the 1st turnpoint then it is a start
+                if point.getAttribute("Exit") == "true":
+                    kind = "start_exit"
+                else:
+                    kind = "start_enter"
+            else:
+                if point == tpoints[-1]:   # if it is the last turnpoint i.e. the goal
+                    if point.getAttribute("type") == "line":
+                        kind = "goal_cylinder"    # TODO(kuaka): change to 'line' once we can process it 
+                    else:
+                        kind = "goal_cylinder"
+                else:
+                    kind = "cylinder"  # All turnpoints other than the 1st and the last are "cylinders".
+                                       # In theory they could be "End_of_speed_section" but this is not supported by LK8000.
+                                       # For paragliders it would be safe to assume that the 2nd to last is always "End_of_speed_section"
+            
+                      
+            turnpoint = Turnpoint(lat, lon, radius, kind)
+            turnpoints.append(turnpoint)
+        task = Task(turnpoints, start_time, end_time)
+        return task
+    
+    
+    def __init__(self, turnpoints, start_time, end_time):
+        self.turnpoints = turnpoints
+        self.start_time = start_time
+        self.end_time = end_time
+        
+        
+        
+    def check_flight(self, Flight):
+        """ Checks a Flight object against the task. 
+            Args:
+                   Flight: a Flight object
+            Returns:
+                    a list of GNSSFixes of when turnpoints were achieved.            
+        """
+        reached_turnpoints = []   
+        proceed_to_start = False
+        t=0
+        for fix in Flight.fixes:
+            if t >= len(self.turnpoints):
+                break  # pilot has arrived in goal (last turnpoint) so we can stop.
+   
```

Comment:
I just realized this function does not use `end_time`. I think what would be sufficient is an `if` statement here:

```
    if end_time < self.turnpoints[t].rawtime:
        # Task has ended
        break